app/src/main/java/com/byoutline/kickmaterial/features/projectlist/ProjectsListFragment.kt

```diff
@@ -22,31 +21,29 @@ import com.byoutline.kickmaterial.utils.KickMaterialFragment
 import com.byoutline.kickmaterial.utils.LUtils
 import com.byoutline.kickmaterial.views.EndlessRecyclerView
 import com.byoutline.secretsauce.activities.showFragment
+import com.byoutline.secretsauce.di.Injectable
+import com.byoutline.secretsauce.di.inflateAndSetViewModel
+import com.byoutline.secretsauce.di.lazyViewModelWithAutoLifecycle
 import timber.log.Timber
-import javax.inject.Inject
+
 
 /**
  * @author Pawel Karczewski <pawel.karczewski at byoutline.com> on 2015-01-03
  */
-class ProjectsListFragment : KickMaterialFragment(), ProjectClickListener, FieldStateListener, EndlessRecyclerView.EndlessScrollListener {
-
-    @Inject
-    lateinit var viewModel: ProjectListViewModel
+class ProjectsListFragment : KickMaterialFragment(), Injectable, ProjectClickListener, FieldStateListener, EndlessRecyclerView.EndlessScrollListener {
 
+    private val viewModel by lazyViewModelWithAutoLifecycle(this, ProjectListViewModel::class)
+    lateinit var binding: FragmentProjectsBinding
```

Comment:
Example of Fragment related changes.
ViewModel will be created lazily on the first use (which in this case in onCreateView during inflate step). 
For performance this lazy method does it initlization without thread safety by default (since we call it from mainThread anyway)